syntax = "proto3";

message Empty {
}

message Statement {
  string type = 1;
}

message Entity {
  string type = 1;
  repeated string values = 2;
}

message EntityResponse {
  repeated Entity entities = 1;
}

message Command {
   string type = 1;
}

message Accuracy {
  float acc = 1;
}

message DataObject {
   string statement = 1;
   string label = 2;
}

service DataService {
  rpc GetData(Empty) returns (DataObject);
}

service MainKG {
  rpc RequestKeywords(Empty) returns (Statement);
  rpc ReceiveKeywords(EntityResponse) returns (Empty);
}

service MainService {
  rpc GetKeywords(Statement) returns (EntityResponse);
  rpc ReceiveData(DataObject) returns (Empty);
}
